docs/platform/self_hosting/configuration.mdx

```diff
@@ -129,61 +233,212 @@ tolgee.smtp.username=*****************
 tolgee.smtp.password=*****************
 tolgee.smtp.port=465
 tolgee.smtp.auth=true
-tolgee.smtp.sslEnabled=true
+tolgee.smtp.ssl-enabled=true
 tolgee.smtp.from=Tolgee <no-reply@tolgee.yourserver.something>
 ```
 
 ### host
-Smtp host
+SMTP host
 ### username
-The username for smtp authentication
+The username for SMTP authentication
 ### password
-Password for smtp authentication
+Password for SMTP authentication
 ### port
+SMTP port
 
 ### auth
 Whether authentication is enabled. (default: `false`)
-### tlsEnabled
-Whether tls is enabled (default: `false`)
-### sslEnabled
-Whether ssl is enabled (default: `false`)
-### tlsRequired
-Whether ssl is required (default: `false`)
+### tls-enabled
+Whether TLS is enabled .(default: `false`)
+### ssl-enabled
+Whether SSL is enabled. (default: `false`)
+### tls-required
+Whether SSL is required. (default: `false`)
 ### from
 The sender name and address in standard SMTP format.
 
-## LDAP authentication properties
-This feature is currently experimental! (prefix: tolgee.authentication.ldap)
- - enabled
- - port
- - urls
- - baseDn
- - securityPrincipal
- - principalPassword
- - userDnPattern
+## Machine translation
+Prefix: `tolgee.machine-translation`
+
+### free-credits-amount
+Amount of machine translations users of the Free tier can request per month. Used by Tolgee Cloud,
+see [pricing](/pricing). Set to `-1` to disable credit-based limitation. (default: `-1`)
+
+## Google Cloud's Cloud Translation
+Prefix: `tolgee.machine-translation.google`
+
+See [Google Cloud's Cloud Translation](https://cloud.google.com/translate) page for more information and applicable
+pricing.
+
+### default-enabled
+Whether Google-powered machine translation is enabled. (default: `true`)
+
+### default-primary
+Whether to use Google Cloud's Cloud Translation as a primary translation engine. (default: `true`)
+
+### api-key
+Google Cloud API key.
+
+## Amazon Translate
+Prefix: `tolgee.machine-translation.aws`
+
+See [AWS's Amazon Translate](https://aws.amazon.com/translate) page for more information and applicable pricing.
+
+### default-enabled
+Whether AWS-powered machine translation is enabled. (default `true`)
+
+### default-primary
+Whether to use Amazon Translate as a primary translation engine. (default: `false`)
+
+### access-key
+AWS access key.
+
+### secret-key
+AWS secret key.
+
+### region
+AWS region. (default: `eu-central-1`)
+
+## Socket.IO
+Prefix: `tolgee.socket-io`
+
+Tolgee exposes a Socket.IO server to allow you to get realtime data modification events.
+
+### enabled
+Whether the Socket.IO server is enabled. (default: `true`)
+
+### port
+Port the Socket.IO server will listen on. (default: `9090`)
+
+### host
+Host the Socket.IO server will listen on. (default: `0.0.0.0`)
+
+### use-redis
+Whether Tolgee should use Redis to store clients. Useful if you have multiple instances of Tolgee for load balancing.
```

Comment:
It doesn't store the clients and it's not useful for load balancing. We use Redis for different 3 things. 
1. Caching
2. Pub/Sub - so the websocket events are sent by all instances
3. Distributed locking - We use this for rate limits. When we have multiple instances and we want to store how many requests were originated from one IP or user, we cannot write to redis in the same time, since we would miss some requests. It would basically increment the rate limit buckets inconsistently when the requests hit different instances. So we lock the resource, write and unlock.

Reply:
Well, in this context specifically the use of Redis applies to the pub/sub server, which is why I mentioned clients. As I understood the inner-work of the underlying Socket.IO implementation you're using, the library uses Redis to keep track of connected clients (what I meant by "store clients") to dispatch events to everyone, including people connected to another node. The mention of load-balancing is just as an example of why people may run multiple Tolgee instances.

I think it just needs better wording there

Reply:
Oh right. Got how you meant it.